include/grpcpp/alts_context.h

```diff
@@ -0,0 +1,75 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_ALTS_CONTEXT_H
+#define GRPCPP_ALTS_CONTEXT_H
+
+#include <grpcpp/impl/codegen/security/auth_context.h>
+
+#include <memory>
+// might need forward declaration
+#include "src/core/tsi/alts/handshaker/alts_tsi_handshaker.h"
+#include "src/proto/grpc/gcp/altscontext.upb.h"
+
+namespace grpc {
+
+enum SecurityLevel {
+  SECURITY_NONE = 0,
+  INTEGRITY_ONLY = 1,
+  INTEGRITY_AND_PRIVACY = 2
+};
+
+typedef struct Versions {
+  int major_version;
+  int minor_version;
+} Versions;
+
+typedef struct RpcProtocolVersions {
+  Versions max_rpc_versions;
+  Versions min_rpc_versions;
+} RpcProtocolVersions;
+
+class AltsContext {
+ public:
+  explicit AltsContext(const AuthContext& auth_context);
```

Comment:
Given AuthContext, it does not necessarily have AltsContext. 

My suggestion would be
1. AltsContext class takes grpc_gcp_AltsContext as constructor
2. Use 
```
std::unique_ptr<AltsContext> GetAltsContextFromAuthContext(
    const AuthContext& auth_context);
```
If there is AltsContext inside AuthContext, it will return a unique_ptr on the object, otherwise, return nullptr